### ISSUE TITLE
xmllint crashes on SublimeLinter 4

### ISSUE THREAD

**Issue by bmix** [SublimeLinter/SublimeLinter-xmllint#9]
I understand, that the byline for this Linter states, that it is for *SublimeLinter 3*. Since it works on many files, however, I assume, it may be, that this remark has not yet been updated?

Since it may aid in updating to v4 I would like to post the debug-log (please note, that I have some paths symlinked from C:\Users\bmix\ to S:\Users\bmix\, just in case this causes some head scratching, see also https://github.com/SublimeLinter/SublimeLinter/issues/1179):

```
reloading plugin SublimeLinter.lint
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.lint
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin SublimeLinter.tooltips_view
reloading plugin SublimeLinter-annotations.linter
reloading plugin SublimeLinter-contrib-htmlhint.linter
reloading plugin SublimeLinter-csslint.linter
reloading plugin SublimeLinter-html-tidy.linter
reloading plugin SublimeLinter-jshint.linter
reloading plugin SublimeLinter-json.linter
reloading plugin SublimeLinter-xmllint.linter
Unable to open /C/Users/bmix/AppData/Roaming/Sublime Text 3/Packages/SublimeLinter/SublimeLinter.sublime-settings
reloading settings Packages/User/SublimeLinter.sublime-settings
SublimeLinter: SublimeLinter.sublime-package:71: Logging installed; log level INFO
SublimeLinter: #7 SublimeLinter.sublime-package:782: 'annotations' is linting 'SublimeLinter.sublime-settings'
SublimeLinter: #8 SublimeLinter.sublime-package:782: 'json' is linting 'SublimeLinter.sublime-settings'
SublimeLinter: SublimeLinter.sublime-package:103: PATH:
    C:\Cygwin\bin
    C:\Users\bmix\AppData\Roaming\npm\node_modules
    C:\WINDOWS\system32
    C:\WINDOWS
    C:\WINDOWS\System32\Wbem
    C:\Python35\
    C:\Python35\Scripts\
    C:\Python27\
    C:\Python27\Scripts\
    C:\ProgramData\Oracle\Java\javapath
    S:\Applications\D\dmd2\windows\bin
    C:\Program Files\Java\jdk-9.0.1\bin\
    C:\ProgramData\chocolatey\bin
    C:\Program Files (x86)\Verschiedenes
    S:\Applications\pandoc
    C:\Program Files (x86)\FreeArc\bin
    C:\Program Files (x86)\FreeArc\PowerPack\bin
    C:\Program Files (x86)\Sudowin\Clients\Console
    C:\Program Files\TortoiseSVN\bin
    C:\Program Files\Calibre2\
    C:\Program Files (x86)\Calibre2\
    C:\Program Files (x86)\Tesseract-OCR
    C:\Program Files\nodejs\
    C:\WINDOWS\System32\WindowsPowerShell\v1.0\
    C:\Program Files (x86)\BaseX\bin
    C:\Program Files\ooRexx
    C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\
    C:\Program Files\Microsoft SQL Server\110\Tools\Binn\
    C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\
    C:\Program Files\Microsoft SQL Server\120\Tools\Binn\
    C:\Program Files (x86)\ActiveState Komodo IDE 8\
    C:\Program Files (x86)\QuickTime Alternative\QTSystem
    C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common
    C:\PROGRA~1\MATROX~1\System32
    C:\PROGRA~1\MATROX~1\System64
    C:\Program Files\Common Files\Microsoft Shared\Windows Live
    C:\Program Files\TortoiseGit\bin
    C:\Program Files (x86)\PuTTY\
    C:\Program Files\Git\cmd
    C:\Cygwin\bin
    C:\Users\bmix\bin
    C:\Program Files (x86)\apache-ant\bin
    C:\Users\bmix\AppData\Roaming\npm
    S:\Users\bmix\.cargo\bin
    C:\Users\bmix\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0
    C:\Users\bmix\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\bin
    C:\Users\bmix\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\WinSDK\Bin
    C:\Program Files (x86)\calumet\bin
    C:\Users\bmix\AppData\Local\Android\android-studio\sdk\platform-tools
    C:\Users\bmix\AppData\Local\Android\android-studio\sdk\tools
    C:\xulrunner
    C:\Program Files (x86)\ActiveState Komodo IDE 8\
    C:\Users\bmix\AppData\Local\Microsoft\WindowsApps
    C:\Program Files\Oracle\VirtualBox
    C:\Program Files (x86)\Verschiedenes\xmlsh_1_2_5
    C:\Program Files (x86)\Scapple
    C:\Program Files (x86)\AmiKit
    S:\Applications\chrlauncher-win32-stable-codecs-sync\bin
SublimeLinter: SublimeLinter.sublime-package:256: detected syntax: xml
SublimeLinter: #9 SublimeLinter.sublime-package:782: 'xmllint' is linting 'filetypes.xml'
SublimeLinter: #9 SublimeLinter.sublime-package:1186: xmllint: filetypes.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #9 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\TheDesktop
SublimeLinter: #9 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
SublimeLinter: #9 SublimeLinter.sublime-package:838: xmllint: No matches for regex: ^.+?:(?P<line>\d+):.+?: (?P<message>[^\r\n]+)\r?\n[^\r\n]*\r?\n(?P<col>[^\^]*)\^
SublimeLinter: SublimeLinter.sublime-package:256: detected syntax: xslt
SublimeLinter: SublimeLinter.sublime-package:256: detected syntax: xml
SublimeLinter: #10 SublimeLinter.sublime-package:782: 'xmllint' is linting 'xproc.xpl'
SublimeLinter: #10 SublimeLinter.sublime-package:1186: xmllint: xproc.xpl ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #10 SublimeLinter.sublime-package:1189: xmllint: cwd: C:\Users\bmix\AppData\Roaming\Microsoft\Windows\Templates
SublimeLinter: #10 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
SublimeLinter: #10 SublimeLinter.sublime-package:838: xmllint: No matches for regex: ^.+?:(?P<line>\d+):.+?: (?P<message>[^\r\n]+)\r?\n[^\r\n]*\r?\n(?P<col>[^\^]*)\^
SublimeLinter: SublimeLinter.sublime-package:256: detected syntax: xml
SublimeLinter: #11 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #11 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #11 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #12 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #13 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #11 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
    -:2: namespace warning : xmlns: URI mydummynsdomain.org/ns/notes is not absolute
    <notes xmlns="mydummynsdomain.org/ns/notes"
                                           ^
    -:19: namespace error : Namespace prefix nml for role on date is not defined
        <dc:date nml:role="exported">2017-11-12T05-11+0100</dc:date>
                                    ^
SublimeLinter: #11 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 821, in lint
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 902, in process_match
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 79, in full_line
IndexError: list index out of range
SublimeLinter: #14 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #14 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #14 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #15 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #16 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: Buffer 42 inconsistent. Aborting lint. 
SublimeLinter: Buffer 42 inconsistent. Aborting lint. 
SublimeLinter: Buffer 42 inconsistent. Aborting lint. 
SublimeLinter: #17 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #17 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #17 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #19 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #18 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #17 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
    -:2: namespace warning : xmlns: URI httpmydummynsdomain.org/ns/notes is not absolute
    <notes xmlns="httpmydummynsdomain.org/ns/notes"
                                               ^
    -:19: namespace error : Namespace prefix nml for role on date is not defined
        <dc:date nml:role="exported">2017-11-12T05-11+0100</dc:date>
                                    ^
SublimeLinter: #17 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 821, in lint
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 902, in process_match
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 79, in full_line
IndexError: list index out of range
SublimeLinter: #20 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #20 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #20 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #21 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #22 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #20 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
    -:19: namespace error : Namespace prefix nml for role on date is not defined
        <dc:date nml:role="exported">2017-11-12T05-11+0100</dc:date>
                                    ^
SublimeLinter: #20 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 821, in lint
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 902, in process_match
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 79, in full_line
IndexError: list index out of range
SublimeLinter: #23 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #23 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #23 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #24 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #25 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #23 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
    -:19: namespace error : Namespace prefix nml for role on date is not defined
        <dc:date nml:role="exported">2017-11-12T05-11+0100</dc:date>
                                    ^
SublimeLinter: #23 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 821, in lint
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 902, in process_match
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 79, in full_line
IndexError: list index out of range
SublimeLinter: #26 SublimeLinter.sublime-package:782: 'xmllint' is linting 'NOTEML.xml'
SublimeLinter: #26 SublimeLinter.sublime-package:1186: xmllint: NOTEML.xml ['C:\\Cygwin\\bin\\xmllint.exe', '--noout', '-']
SublimeLinter: #26 SublimeLinter.sublime-package:1189: xmllint: cwd: S:\Users\bmix\Eigene Projekte\XML
SublimeLinter: #27 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #28 SublimeLinter.sublime-package:782: 'annotations' is linting 'NOTEML.xml'
SublimeLinter: #26 SublimeLinter.sublime-package:809: xmllint output:
    file:/c/Users/bmix/AppData/Roaming/Microsoft/Windows/Templates/xmlcatalog/catalog.xml:99: element uri: Catalog error : uri entry 'uri' broken ?: file://C:/Users/bmix/lib/uuid.xsl
    -:19: namespace error : Namespace prefix nml for role on date is not defined
        <dc:date nml:role="exported">2017-11-12T05-11+0100</dc:date>
                                    ^
SublimeLinter: #26 __init__.py:1273: ERROR: Linter crashed.

Traceback (most recent call last):
  File "lint.backend in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 821, in lint
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 902, in process_match
  File "lint.linter in C:\Users\bmix\AppData\Roaming\Sublime Text 3\Installed Packages\SublimeLinter.sublime-package", line 79, in full_line
IndexError: list index out of range
```

**Comment by kaste** [comment]
Thanks for reporting. Not sure if we resolve that here or in SublimeLinter core.

**Comment by kaste** [comment]
Basically the regex we have to 'parse' the output is not good enough. But I'm not that good at regexes. Can we change the output format to something easier?

Can you fix the regex?

**Comment by kaste** [comment]
Proposed fix in #10
